workspace/workspace.go

```diff
@@ -102,3 +103,57 @@ func (ws Workspace) Locate(exercise string) ([]string, error) {
 	}
 	return paths, nil
 }
+
+// SolutionPath returns the full path where the exercise will be stored.
+// By default this the directory name matches that of the exercise, but if
+// a different solution already exists, then a numeric suffix will be added
+// to the name.
+func (ws Workspace) SolutionPath(exercise, solutionID string) (string, error) {
+	paths, err := ws.Locate(exercise)
+	if !IsNotExist(err) && err != nil {
+		return "", err
+	}
+
+	return ws.DetermineSolutionPath(paths, exercise, solutionID, PathContainsSolution)
+}
+
+// TODO: PathContainsSolution might need a different name.
+func PathContainsSolution(path, solutionID string) (bool, error) {
+	s, err := NewSolution(path)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return s.ID == solutionID, nil
+}
+
+// TODO: DetermineSolutionPath might need a different name.
```

Comment:
I'm already using `SolutionPath` at the top level, which is why I suspect that my abstraction is a little bit forced. When it's hard to name things that tends to be a sign that my thinking isn't completely clear yet.

I like `ResolveSolutionPath` better than `DetermineSolutionPath`. I think I'll go with that.